### PR TITLE
SAK-32661 Initial commit of Noodle's MyCalendar widget

### PR DESCRIPTION
**Pull request by adrianfish**

### CODE REVIEW

**Comment by adrianfish** [comment]
screen shot 2017-07-07 at 07 59 17
screen shot 2017-07-07 at 07 59 44
screen shot 2017-06-30 at 08 06 55

File: library/src/morpheus-master/sass/modules/tool/mycalendar/_mycalendar.scss [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        justify-content: center;
+        align-items: center;
+        color: #777;
+    }
+    .Mrphs-sakai-mycalendar .ui-datepicker-prev
+    , .Mrphs-sakai-mycalendar .ui-datepicker-next {
+        position: relative !important;
+        left: 0px;
+        top: 0px;
+        width: 24px;
```
Comment (ottenhoff) [inline library/src/morpheus-master/sass/modules/tool/mycalendar/_mycalendar.scss]: Do you have to use important? "0" is preferred over "0px"

File: library/src/morpheus-master/sass/modules/tool/mycalendar/_mycalendar.scss [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        order: 2;
+    }
+    .Mrphs-sakai-mycalendar .ui-datepicker-next {
+        order: 3;
+        margin-left: 15px;
+    }
+
+    .Mrphs-sakai-mycalendar .ui-corner-all {
+        border: none;
+        border-radius: 0 0 0 0;
```
Comment (ottenhoff) [inline library/src/morpheus-master/sass/modules/tool/mycalendar/_mycalendar.scss]: "0" is preferred over "0 0 0 0"

File: library/src/morpheus-master/sass/modules/tool/mycalendar/_mycalendar.scss [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    }
+
+    .mycalendar-event-time {
+        width: 80px;
+    }
+
+    .mycalendar-event-type {
+        font-weight: bold;
+    }
+    .Mrphs-container.Mrphs-sakai-mycalendar {
```
Comment (ottenhoff) [inline library/src/morpheus-master/sass/modules/tool/mycalendar/_mycalendar.scss]: I believe all reference to "Mrphs" should be replaced with the namespace variable?

File: library/src/morpheus-master/sass/modules/tool/mycalendar/_mycalendar.scss [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+    table.mycalendar-events {
+        margin: 0 auto;
+    }
+
+    table.mycalendar-event {
+        margin: 0 auto;
+    }
+
+    @media screen and (min-width: 1100px) {
```
Comment (ottenhoff) [inline library/src/morpheus-master/sass/modules/tool/mycalendar/_mycalendar.scss]: I believe there are variables to handle this .....

File: mycalendar/pom.xml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    </parent>
+	<packaging>war</packaging>
+	<dependencies>
+		<!-- third party dependencies -->
+		<dependency>
+			<groupId>org.apache.wicket</groupId>
+            <artifactId>wicket</artifactId>
+            <version>6.18.0</version>
+			<type>pom</type>
+		</dependency>
```
Comment (ottenhoff) [inline mycalendar/pom.xml]: Can you fix tabs/spaces?

**Comment by adrianfish** [comment]
@ottenhoff I've addressed all your points, I think. Tested all the importants and got rid of most of them.

File: mycalendar/src/webapp/scripts/mycalendar.js [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+        currentSiteId = $(".mycalendar-data").attr("data-siteid");
+        timeZone = $(".mycalendar-data").attr("data-tz");
+        
+        //get all events for the site for this month
+        var todaysDate = moment().format('YYYY-MM-DD');
+        eventList = getEvents(todaysDate);
+
+        var init = function () {
+        
+                $(".mycalendar").datepicker({
```
Comment (ottenhoff) [inline mycalendar/src/webapp/scripts/mycalendar.js]: What datepicker is this invoking?

Our localized Sakai datepicker I believe is invoked with localDatepicker

Comment (adrianfish) [inline mycalendar/src/webapp/scripts/mycalendar.js]: Pretty sure it's jquery ui's datepicker

Comment (ottenhoff) [inline mycalendar/src/webapp/scripts/mycalendar.js]: We have a localized datepicker in use across Sakai.... should be used in nearly every tool now .... announcements/assignments/etc

Comment (adrianfish) [inline mycalendar/src/webapp/scripts/mycalendar.js]: But, it's just a datepicker, right? In this case we are using the date picker as an actual calendar, with events showing as highlighted and the ability to navigate the months and days. The localised one's based on datepicker, so it should be possible to adapt the code.
@mgolden-noodle  ?

Comment (ottenhoff) [inline mycalendar/src/webapp/scripts/mycalendar.js]: If you are just picking dates and not date+times, this one might be fine. have you tested in european locales? Does the date picker use the correct format for the locale? IF yes, let's proceed.

Comment (adrianfish) [inline mycalendar/src/webapp/scripts/mycalendar.js]: Just dates, no times. Mainly the use case is just loading up events from the calendar api and sticking them in a navigable calendar widget. I'll do a bit of testing in spanish tonight and see how the localisation's working.

**Comment by mgolden-noodle** [comment]
I was the one who made sure that the localization works, as it didn't originally. Aside from localizing the calendar itself, it also needs to do the display of the events themselves below - dates and time. This is does work.

I agree with Adrian's view that this is a calendar and not actually a datepicker in the normal sense. Does the synoptic calendar use the datepicker code?

**Comment by steveswinsburg** [comment]
Whilst this uses jQuery datepicker, it is a calendar view, not an actual datepicker. @amiedavis and I wrote the original version for Noodle.

**Comment by ottenhoff** [comment]
Okay this is looking good in my testing. Comments:

1. mycalendar needs to be added to root pom.xml
2. instructors shouldn't be allowed to add as a separate tool, right?
3. can modify default course site to use this widget instead of old Calendar synoptic tool?

**Comment by adrianfish** [comment]
@ottenhoff Don't see why we can't modify the default course site and swap this in. Need some testing though.

**Comment by adrianfish** [comment]
I don't know how to block adding it as a separate tool, though. Just remove project, course etc from the tool xml?

**Comment by ottenhoff** [comment]
> I don't know how to block adding it as a separate tool, though. Just remove project, course etc from the tool xml?

I think so .... just want this to be like the synoptic tools, right? I would just mimic one of those tools

**Comment by adrianfish** [comment]
@ottenhoff Updated the top pom and took out the various site types from the tool xml.

**Comment by adrianfish** [comment]
@ottenhoff And tested it works :)

File: mycalendar/pom.xml [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+            <artifactId>sakai-kernel-util</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>commons-lang</groupId>
+            <artifactId>commons-lang</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>commons-logging</groupId>
+            <artifactId>commons-logging</artifactId>
+        </dependency>
```
Comment (ottenhoff) [inline mycalendar/pom.xml]: I thought we stopped using commons-logging?

Comment (adrianfish) [inline mycalendar/pom.xml]: The code was using it - I switched it to slf4j but forgot the pom.

Comment (adrianfish) [inline mycalendar/pom.xml]: @ottenhoff Pushed.

**Pull request merged**